minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -600,6 +602,97 @@ public synchronized void start() throws IOException, InterruptedException {
     if (executor == null) {
       executor = Executors.newSingleThreadExecutor();
     }
+
+    verifyUp();
+
+  }
+
+  private void verifyUp() throws InterruptedException, IOException {
+
+    Objects.requireNonNull(getClusterControl().managerProcess, "Error starting Manager");
+    Objects.requireNonNull(getClusterControl().managerProcess.info().startInstant().get(),
+        "Error starting Manager");
+
+    Objects.requireNonNull(getClusterControl().gcProcess, "Error starting GC");
+    Objects.requireNonNull(getClusterControl().gcProcess.info().startInstant().get(),
+        "Error starting GC");
+
+    int tsExpectedCount = 0;
+    for (Process tsp : getClusterControl().tabletServerProcesses) {
+      tsExpectedCount++;
+      Objects.requireNonNull(tsp, "Error starting TabletServer");
+      Objects.requireNonNull(tsp.info().startInstant().get(), "Error starting Manager");
+    }
+
+    String zk = config.getExistingZooKeepers() != null ? config.getExistingZooKeepers()
+        : config.getZooKeepers();
+
+    ZooReaderWriter zrw =
+        new ZooReaderWriter(zk, 30000, Property.INSTANCE_SECRET.getDefaultValue());
+
+    String instanceId = null;
+    try {
+      for (String name : zrw.getChildren(Constants.ZROOT + Constants.ZINSTANCES)) {
+        if (name.equals(config.getInstanceName())) {
+          String instanceNamePath = Constants.ZROOT + Constants.ZINSTANCES + "/" + name;
+          byte[] bytes = zrw.getData(instanceNamePath);
+          instanceId = new String(bytes, UTF_8);
+        }
+      }
+    } catch (KeeperException e) {
+      throw new RuntimeException("Unable to read instance name from zookeeper.", e);
+    }
+    String rootPath = ZooUtil.getRoot(instanceId);
```

Comment:
Is `instanceID` only expected to be found once? If so, can we break out of the loop after it is found? 

Also, is it possible that it is never found and `instanceID` stays null? If so, would that cause issues with `ZooUtil.getRoot(instanceId)`?

Reply:
Addressed in 35efd1c